docs/examples/frameworks/pytorch/pytorch-lightning.ipynb

```diff
@@ -0,0 +1,488 @@
+      "I1014 17:16:44.541009 139719591925568 distributed.py:49] GPU available: True, used: True\n",
+      "TPU available: False, using: 0 TPU cores\n",
+      "I1014 17:16:44.543509 139719591925568 distributed.py:49] TPU available: False, using: 0 TPU cores\n",
+      "Using environment variable NODE_RANK for node rank (0).\n",
+      "I1014 17:16:44.545432 139719591925568 distributed.py:49] Using environment variable NODE_RANK for node rank (0).\n",
+      "LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]\n",
+      "I1014 17:16:44.547166 139719591925568 accelerator_connector.py:333] LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]\n",
+      "\n",
+      "  | Name    | Type   | Params\n",
+      "-----------------------------------\n",
+      "0 | layer_1 | Linear | 100 K \n",
+      "1 | layer_2 | Linear | 33 K  \n",
+      "2 | layer_3 | Linear | 2 K   \n",
+      "I1014 17:16:44.609724 139719591925568 lightning.py:1215] \n",
+      "  | Name    | Type   | Params\n",
+      "-----------------------------------\n",
+      "0 | layer_1 | Linear | 100 K \n",
+      "1 | layer_2 | Linear | 33 K  \n",
+      "2 | layer_3 | Linear | 2 K   \n"
+     ]
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "c88a3c2ba1634cf0b89f6a188079d7e1",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "HBox(children=(HTML(value='Training'), FloatProgress(value=1.0, bar_style='info', layout=Layout(flex='2'), max…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\n"
+     ]
+    },
+    {
+     "data": {
+      "text/plain": [
+       "1"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "model = DALILitMNIST()\n",
+    "trainer = Trainer(gpus=1, distributed_backend=\"ddp\", max_epochs=5)\n",
+    "trainer.fit(model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now let us provide the custom DALI iterator wrapper so we don't have to do any extra processing inside `LitMNIST.process_batch`, also PyTorch can learn how big is the dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class BetterDALILitMNIST(LitMNIST):\n",
+    "    def __init__(self):\n",
+    "        super().__init__()\n",
+    "    \n",
+    "    def prepare_data(self):\n",
+    "        device_id = self.local_rank\n",
+    "        shard_id = self.global_rank\n",
+    "        num_shards = self.trainer.world_size\n",
+    "        mnist_pipeline = MnistPipeline(BATCH_SIZE, device='cpu', device_id=device_id, shard_id=shard_id, num_shards=num_shards, num_threads=8)\n",
+    "\n",
+    "        class LightingWrapper(DALIClassificationIterator):\n",
```

Comment:
There are several typos Lighting -> Lightning. Please search and replace.

Reply:
Done